backend/handlers.go

```diff
@@ -42,15 +41,41 @@ func (cs *server) cssHandler(w http.ResponseWriter, r *http.Request, _ httproute
 // the received message.
 func (cs *server) publishHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	body := http.MaxBytesReader(w, r.Body, 8192)
-	msg, err := io.ReadAll(body)
-	if err != nil {
-		http.Error(w, http.StatusText(http.StatusRequestEntityTooLarge), http.StatusRequestEntityTooLarge)
+
+	// Unmarshal the JSON message into a map
+	decoder := json.NewDecoder(body)
+
+	// Create a new map to store the sensorDataMap
+	var sensorIDMap map[string]string
+	// Decode into the sensorDataMap
+	if err := decoder.Decode(&sensorIDMap); err != nil {
+		http.Error(w, "JSON decoding error: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Get the sensorID from the map
+	sensorID, exists := sensorIDMap["sensorID"]
+	if !exists {
+		http.Error(w, "Sensor Id does not exists.", http.StatusBadRequest)
```

Comment:
Don't capitalize and use full stops in error messages. We often use them with string formatting.